changefeedccl: warn when using https sink urls

HonoreDB opened the pull request:
Changefeeds previously used http as their URI to align with backups.
Backups no longer support http, and for changefeeds http is ambiguous,
leading users to create cloudstorage sinks when they meant webhook sinks.
This PR changes the scheme to file-http. Webhooks still use webhook-http.

Fixes #98719.

Release note (enterprise change): Changefeeds that create files over an http connection may now be specified via `INTO 'file-https://'` to disambiguate with `webhook-https`.

HonoreDB committed:
changefeedccl: warn on when using https sink urls

Fixes #98719.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107572)
<!-- Reviewable:end -->

miretskiy commented on pkg/ccl/changefeedccl/changefeed_stmt.go:
```diff
@@ -904,6 +904,15 @@ func validateSink(
+		return err
+	}
+
+	if u.Scheme == changefeedbase.SinkSchemeCloudStorageHTTP || u.Scheme == changefeedbase.SinkSchemeCloudStorageHTTPS {
```
Do we want to do that?
Are we going to bother people who want http with needles warning?

Would it be better to add file-http(s) scheme and prepare http(s) scheme for deprecation?

@amruss ?

HonoreDB committed:
changefeedccl: prepend file- to cloudstorage http(s) schemes

Changefeeds previously used http as their URI to align with backups.
Backups no longer support http, and for changefeeds http is ambiguous,
leading users to create cloudstorage sinks when they meant webhook sinks.
This commit changes the scheme to file-http. Webhooks still use webhook-http.

Informs: https://github.com/cockroachdb/cockroach/issues/98719

Release note (enterprise change): Changefeeds that create files over an http connection may now be specified via `INTO 'file-https://'` to disambiguate with `webhook-https`.

miretskiy commented on pkg/ccl/changefeedccl/changefeed_stmt.go:
```diff
@@ -904,6 +904,18 @@ func validateSink(
+
+	if u.Scheme == changefeedbase.DeprecatedSinkSchemeHTTP || u.Scheme == changefeedbase.DeprecatedSinkSchemeHTTP {
+		p.BufferClientNotice(ctx, pgnotice.Newf(
+			`%s sinks will emit using cloud storage semantics. For a webhook sink, prepend webhook- to the sink URI. Prepend file- to remove this notice.`,
```
maybe rephrase a bit to avoid using "prepend"? Something like "Use webhook-https for webhook semantics, use file-http:// .."

HonoreDB replied:
Done.

miretskiy reviewed (approved):
Please update release note in commit and in PR to indicate deprecation.

HonoreDB commented:
> Please update release note in commit and in PR to indicate deprecation.

Release note's in the second commit, merged it into the PR description.

HonoreDB commented:
bors r=[miretskiy]
